Indicators/ResetCompositeIndicator.cs

```diff
@@ -0,0 +1,88 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using System;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// Class that extends CompositeIndicator to execute a given action once is reset
+    /// </summary>
+    public class ResetCompositeIndicator : IndicatorBase
+    {
+        /// <summary>
+        /// Action to execute once the given CompositeIndicator is reset
+        /// </summary>
+        private Action _extraResetAction;
+
+        /// <summary>
+        /// The CompositeIndicator that will fire the given action once is reset
+        /// </summary>
+        private CompositeIndicator _compositeIndicator;
+
+        /// <summary>
+        /// Gets current value of the given CompositeIndicator
+        /// </summary>
+        public IndicatorDataPoint Current => _compositeIndicator.Current;
+
+        /// <summary>
+        /// Gets a flag indicating when the given CompositeIndicator is ready
+        /// </summary>
+        public override bool IsReady => _compositeIndicator.IsReady;
```

Comment:
It should also override the remaining `CompositeIndicator` properties so that the users can access the inner composite indicator instance properties.